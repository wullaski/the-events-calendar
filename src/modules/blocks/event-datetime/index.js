/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { get } from 'lodash';

/**
 * Internal dependencies
 */
import EventDateTime from './container';
import { BlockIcon } from '@moderntribe/common/elements';
import { date, globals } from '@moderntribe/common/utils';

/**
 * Module Code
 */

const timeZone = get( globals.config(), 'timeZone', {} );

export default {
	id: 'event-datetime',
	title: __( 'Event Date Time', 'events-gutenberg' ),
	description: __(
		'Define the date, time, and duration for your event.',
		'events-gutenberg'
	),
	icon: BlockIcon,
	category: 'tribe-events',
	keywords: [ 'event', 'events-gutenberg', 'tribe' ],

	supports: {
		html: false,
	},

	attributes: {
		start: {
			type: 'string',
			source: 'meta',
			meta: '_EventStartDate',
		},
		end: {
			type: 'string',
			source: 'meta',
			meta: '_EventEndDate',
		},
		allDay: {
			type: 'boolean',
			source: 'meta',
			meta: '_EventAllDay',
		},
		timeZone: {
			type: 'string',
			source: 'meta',
			meta: '_EventTimezone',
		},
		separatorDate: {
			type: 'string',
			source: 'meta',
			meta: '_EventDateTimeSeparator',
		},
		separatorTime: {
			type: 'string',
			source: 'meta',
			meta: '_EventTimeRangeSeparator',
		},
		showTimeZone: {
			type: 'boolean',
			default: get( timeZone, 'show_time_zone', false ),
		},
		timeZoneLabel: {
			type: 'string',
			default: get( timeZone, 'label', date.FORMATS.TIMEZONE.string ),
		},
		// Only Available for classic users
		cost: {
			type: 'string',
			source: 'meta',
			meta: '_EventCost',
		},
		currencySymbol: {
			type: 'string',
			source: 'meta',
			meta: '_EventCurrencySymbol',
		},
		currencyPosition: {
			type: 'string',
			source: 'meta',
			meta: '_EventCurrencyPosition',
		},
	},

	edit: EventDateTime,

	save( props ) {
		return null;
	},
};
