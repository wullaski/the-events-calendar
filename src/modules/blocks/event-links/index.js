/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import EventLinks from './container';
import { BlockIcon } from '@moderntribe/common/elements';

/**
 * Module Code
 */
export default {
	id: 'event-links',
	title: __( 'Event Sharing', 'events-gutenberg' ),
	description: __(
		'Encourage visitors to add your event to their calendars with handy sharing buttons.',
		'events-gutenberg'
	),
	icon: BlockIcon,
	category: 'tribe-events',
	keywords: [ 'event', 'events-gutenberg', 'tribe' ],

	supports: {
		html: false,
	},

	attributes: {
		googleCalendarLabel: {
			type: 'html',
			default: __( 'Google Calendar', 'events-gutenberg' ),
		},
		iCalLabel: {
			type: 'html',
			default: __( 'iCal Export', 'events-gutenberg' ),
		},
		hasiCal: {
			type: 'html',
			default: true,
		},
		hasGoogleCalendar: {
			type: 'html',
			default: true,
		},
	},

	edit: EventLinks,
	save( props ) {
		return null;
	},
};

