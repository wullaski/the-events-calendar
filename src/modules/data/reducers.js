/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import blocks from './blocks';
import ui from './ui';
import search from './search';
import details from './details';
import forms from './forms';

export default combineReducers( {
	blocks,
	ui,
	search,
	details,
	forms,
} );

