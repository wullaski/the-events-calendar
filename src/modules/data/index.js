/**
 * External dependencies
 */
import reducer from './reducers';

import { actions } from '@moderntribe/common/data/reducers/plugins';
import { store } from '@moderntribe/common/store';

export const initStore = () => {
	const { dispatch, injectReducers } = store;

	dispatch( actions.addPlugin( 'events' ) );
	injectReducers( { events: reducer } );
};

export const getStore = () => store;
