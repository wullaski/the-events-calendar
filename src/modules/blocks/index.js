import { registerBlockType } from '@wordpress/blocks';

import classicEventDetails from '@moderntribe/events/blocks/classic-event-details';

import eventDateTime from '@moderntribe/events/blocks/event-datetime';
import eventVenue from '@moderntribe/events/blocks/event-venue';
import eventOrganizer from '@moderntribe/events/blocks/event-organizer';
import eventLinks from '@moderntribe/events/blocks/event-links';
import eventPrice from '@moderntribe/events/blocks/event-price';
import eventCategory from '@moderntribe/events/blocks/event-category';
import eventTags from '@moderntribe/events/blocks/event-tags';
import eventWebsite from '@moderntribe/events/blocks/event-website';
import FeaturedImage from '@moderntribe/events/blocks/featured-image';
import { initStore } from '@moderntribe/events/data';

import './style.pcss';

initStore();

const blocks = [
	classicEventDetails,
	eventDateTime,
	eventVenue,
	eventOrganizer,
	eventLinks,
	eventPrice,
	eventCategory,
	eventTags,
	eventWebsite,
	FeaturedImage,
];

blocks.forEach( block => {
	const blockName = `tribe/${ block.id }`;
	registerBlockType( blockName, block );
} );

export default blocks;

