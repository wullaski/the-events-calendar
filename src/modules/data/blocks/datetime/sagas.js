/**
 * External Dependencies
 */
import { put, takeEvery, select, takeLatest, call, all } from 'redux-saga/effects';
import { delay } from 'redux-saga';
import { date as dateUtil } from '@moderntribe/common/src/modules/utils';

/**
 * Internal dependencies
 */
import {
	types,
	selectors,
	actions,
	thunks,
} from '@moderntribe/events/data/blocks/datetime';
import { rangeToNaturalLanguage } from '@moderntribe/common/utils/date';

/**
 * Set the human readable label into the store, based on a start and end date to generate a new label based on those
 *
 * @since 0.3.1-alpha
 *
 * @param {object} dates An object that represents the start / end date
 * @returns {IterableIterator<*>}
 */
export function* setHumanReadableLabel( dates = {} ) {
	const currentLabel = yield select( selectors.getNaturalLanguageLabel );

	if ( currentLabel === '' ) {
		return;
	}

	const updatedLabel = yield call( rangeToNaturalLanguage, dates.start, dates.end );
	yield put( actions.setNaturalLanguageLabel( updatedLabel ) );
}

/**
 * Set the humman readable label from an action that sets the date on the event either the start or end date
 * first selects as default value current start and end dates after that based on the action type selects the
 * date to be replaced or the new date to be set before doing the conversion into text.
 *
 * @since 0.3.1-alpha
 *
 * @param {object} action Dispateched by the component and watched by this generator
 * @returns {IterableIterator<*>}
 */
export function* setHumanReadableFromDate( action ) {
	const dates = {
		start: yield select( selectors.getStart ),
		end: yield select( selectors.getEnd ),
	};

	if ( action.type === types.SET_END_DATE_TIME ) {
		dates.end = action.payload.end;
	} else {
		dates.start = action.payload.start;
	}
	yield call( setHumanReadableLabel, dates );
}

/**
 * Generator used to reset the label of the input using the current values on the start and end date, useful when
 * the user enters a data that is not valid or can't be parsed into a valid date so the input always reflects a valid
 * date based on the current values set into the dates of the event.
 *
 * @since 0.3.1-alpha
 *
 * @returns {IterableIterator<*>}
 */
export function* resetNaturalLanguageLabel() {
	const dates = yield all( {
		start: select( selectors.getStart ),
		end: select( selectors.getEnd ),
	} );
	yield call( setHumanReadableLabel, dates );
}

/**
 * Fired when the Human Readable has a change() event fired on the input, in combination with takeLatest and delay
 * simulates the debounce functionality as gets executed every WAIT_PERIOD_IN_MILLISECONDS to prevent doing work
 * when the input is handling a new change() events
 *
 * @since 0.3.1-alpha
 *
 * @returns {IterableIterator<*>}
 */
export function* onHumanReadableChange() {
	// Wait in case there's a new change on the input
	yield call( delay, 700 );

	const label = yield select( selectors.getNaturalLanguageLabel );
	const dates = dateUtil.labelToDate( label );

	if ( dates.start === null && dates.end === null ) {
		yield call( resetNaturalLanguageLabel );
	} else {
		yield put( thunks.setDateTime( dates ) );
	}
}

/**
 * Watchers of actions and act accordingly to each.
 *
 * @since 0.3.1-alpha
 *
 * @returns {IterableIterator<*|ForkEffect>}
 */
export default function* watchers() {
	yield takeEvery( types.SET_START_DATE_TIME, setHumanReadableFromDate );
	yield takeEvery( types.SET_END_DATE_TIME, setHumanReadableFromDate );
	yield takeLatest( types.SET_NATURAL_LANGUAGE_LABEL, onHumanReadableChange );
}
