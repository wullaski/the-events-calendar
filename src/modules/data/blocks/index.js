/**
 * Internal dependencies
 */
import reducer from './reducer';
import './sagas';

export default reducer;
