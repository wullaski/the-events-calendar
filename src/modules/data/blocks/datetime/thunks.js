/**
 * WordPress dependencies
 */
import { applyFilters } from '@wordpress/hooks';

/**
 * Internal dependencies
 */
import {
	setStartDateTime,
	setEndDateTime,
	setAllDay as setAllDayAction,
	setMultiDay as setMultiDayAction,
	setSeparatorDate,
	setSeparatorTime,
	setTimeZone,
	setTimeZoneLabel,
	setTimeZoneVisibility,
	setNaturalLanguageLabel,
} from './actions';
import { DEFAULT_STATE } from './reducer';
import { maybeBulkDispatch } from '@moderntribe/events/data/utils';
import { date, moment, time } from '@moderntribe/common/utils';

const {
	isSameDay,
	parseFormats,
	replaceDate,
	setTimeInSeconds,
	toDateTime,
	toMoment,
	adjustStart,
} = moment;

export const setStartTime = ( { start, seconds } ) => ( dispatch ) => {
	const startDateTime = toDateTime( setTimeInSeconds( toMoment( start ), seconds ) );
	dispatch( setStartDateTime( startDateTime ) );
};

export const setEndTime = ( { end, seconds } ) => ( dispatch ) => {
	const endDateTime = toDateTime( setTimeInSeconds( toMoment( end ), seconds ) );
	dispatch( setEndDateTime( endDateTime ) );
};

export const setAllDay = ( { start, end, isAllDay } ) => ( dispatch ) => {
	if ( isAllDay ) {
		const startDateTime = toDateTime( setTimeInSeconds( toMoment( start ), 0 ) );
		const endDateTime = toDateTime( setTimeInSeconds( toMoment( end ), time.DAY_IN_SECONDS - 1 ) );
		dispatch( setStartDateTime( startDateTime ) );
		dispatch( setEndDateTime( endDateTime ) );
	}

	dispatch( setAllDayAction( isAllDay ) );
}

export const setDates = ( { start, end, from, to } ) => ( dispatch ) => {
	const startMoment = toMoment( start );
	const endMoment = toMoment( end );

	const result = adjustStart(
		replaceDate( startMoment, toMoment( from ) ),
		replaceDate( endMoment, toMoment( to || from ) ),
	);

	dispatch( setStartDateTime( toDateTime( result.start ) ) );
	dispatch( setEndDateTime( toDateTime( result.end ) ) );
};

export const setDateTime = ( { start, end } ) => ( dispatch ) => {
	const result = adjustStart(
		toMoment( start ),
		toMoment( end || start ),
	);

	const isMultiDay = ! isSameDay( result.start, result.end );
	dispatch( setStartDateTime( toDateTime( result.start ) ) );
	dispatch( setEndDateTime( toDateTime( result.end ) ) );
	dispatch( setMultiDayAction( isMultiDay ) );
}

export const setMultiDay = ( { start, end, isMultiDay } ) => ( dispatch ) => {
	if ( isMultiDay ) {
		const RANGE_DAYS = applyFilters( 'tec.datetime.defaultRange', 3 );
		const endMoment = toMoment( end ).clone().add( RANGE_DAYS, 'days' );
		dispatch( setEndDateTime( toDateTime( endMoment ) ) );
	} else {
		const startMoment = toMoment( start );
		const result = adjustStart(
			startMoment,
			replaceDate( toMoment( end ), startMoment ),
		);

		dispatch( setStartDateTime( toDateTime( result.start ) ) );
		dispatch( setEndDateTime( toDateTime( result.end ) ) );
	}

	dispatch( setMultiDayAction( isMultiDay ) );
};

export const setInitialState = ( { get, attributes } ) => ( dispatch ) => {
	const timeZone = get( 'timeZone', DEFAULT_STATE.timeZone );
	const defaultTimeZone = get( 'timeZoneLabel', timeZone );

	maybeBulkDispatch( attributes, dispatch )( [
		[ setStartDateTime, 'start', DEFAULT_STATE.start ],
		[ setEndDateTime, 'end', DEFAULT_STATE.end ],
		[ setAllDayAction, 'allDay', DEFAULT_STATE.allDay ],
		[ setSeparatorDate, 'separatorDate', DEFAULT_STATE.dateTimeSeparator ],
		[ setSeparatorTime, 'separatorTime', DEFAULT_STATE.timeRangeSeparator ],
		[ setTimeZone, 'timeZone', DEFAULT_STATE.timeZone ],
		[ setTimeZoneLabel, 'timeZoneLabel', defaultTimeZone ],
		[ setTimeZoneVisibility, 'showTimeZone', DEFAULT_STATE.showTimeZone ],
	] );

	const values = {
		start: DEFAULT_STATE.start,
		end: DEFAULT_STATE.end,
	};

	if ( attributes.start ) {
		values.start = toDateTime( parseFormats( attributes.start ) );
		dispatch( setStartDateTime( values.start ) );
	}

	if ( attributes.end ) {
		values.end = toDateTime( parseFormats( attributes.end ) );
		dispatch( setEndDateTime( values.end ) );
	}

	dispatch( setNaturalLanguageLabel( date.rangeToNaturalLanguage( values.start, values.end ) ) );

	dispatch( setMultiDayAction( ! isSameDay( values.start, values.end ) ) );
};
