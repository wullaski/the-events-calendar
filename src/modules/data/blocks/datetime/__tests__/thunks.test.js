/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';


/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';
import { thunks } from '@moderntribe/events/data/blocks/datetime';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

let initialState = {};
let store;

describe( '[STORE] - Datetime thunks', () => {
	beforeEach( () => {
		initialState = {
			blocks: {
				datetime: {
					start: '2018-06-05 17:00:00',
					end: '2018-06-05 17:30:00',
					dateTimeSeparator: ' @ ',
					timeRangeSeparator: ' - ',
					allDay: false,
					multiDay: false,
					timezone: 'UTC',
				},
			},
		};

		store = mockStore( initialState );
	} );

	test( 'Initial set of state', () => {
		store = mockStore( {} );

		const attributes = {
			start: 'June 5, 2018 5:00 pm',
			end: 'June 5, 2018 5:30 pm',
			dateTimeSeparator: ' @ ',
			timeRangeSeparator: ' - ',
			allDay: false,
			multiDay: false,
			timeZone: 'UTC',
		};

		const get = jest.fn( ( name ) => attributes[ name ] );
		store.dispatch( thunks.setInitialState( { get, attributes } ) );

		expect( get ).toHaveBeenCalled();
		expect( get ).toHaveBeenCalledTimes( 2 );
		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Should set the human readable', () => {
		store = mockStore( {} );
		const attributes = {
			start: '',
			end: '',
		};

		const get = jest.fn( ( name ) => attributes[ name ] );
		store.dispatch( thunks.setInitialState( { get, attributes } ) );

		const types = store.getActions().map( ( action ) => action.type );
		const values = store.getActions().map( ( action ) => action.payload )
			.reduce( ( obj, item ) => {
				return { ...obj, ...item };
			}, {} );
		const expectedActions = [
			`${ PREFIX_EVENTS_STORE }/SET_START_DATE_TIME`,
			`${ PREFIX_EVENTS_STORE }/SET_END_DATE_TIME`,
			`${ PREFIX_EVENTS_STORE }/SET_NATURAL_LANGUAGE_LABEL`,
			`${ PREFIX_EVENTS_STORE }/SET_MULTI_DAY`,
		];
		expect( get ).toHaveBeenCalled();
		expect( get ).toHaveBeenCalledTimes( 2 );
		expect( types ).toEqual( expectedActions );
		expect( values.start ).not.toBe( '' );
		expect( values.end ).not.toBe( '' );
		expect( values.label ).not.toBe( '' );
	} );

	test( 'Set start time', () => {
		const attributes = {
			start: '2018-06-05 17:00:00',
			seconds: 7200,
		};

		store.dispatch( thunks.setStartTime( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set end time', () => {
		Date.now = jest.fn( () => '2018-08-06T05:23:19.000Z' );
		const attributes = {
			end: '2018-06-05 17:30:00',
			seconds: 64800,
		};

		store.dispatch( thunks.setEndTime( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set all day on', () => {
		const attributes = {
			start: '2018-06-05 17:00:00',
			end: '2018-06-05 17:30:00',
			isAllDay: true,
		};

		store.dispatch( thunks.setAllDay( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set all day off', () => {
		const attributes = {
			start: '2018-06-05 00:00:00',
			end: '2018-06-05 23:59:59',
			isAllDay: false,
		};

		store.dispatch( thunks.setAllDay( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set dates from date picker with multi day off', () => {
		const attributes = {
			start: '2018-06-05 17:00:00',
			end: '2018-06-05 17:30:00',
			from: new Date( 'Wed Sep 19 2018 12:00:00' ),
			to: undefined,
		};

		store.dispatch( thunks.setDates( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set dates from date picker with multi day on', () => {
		const attributes = {
			start: '2018-06-05 17:00:00',
			end: '2018-06-05 17:30:00',
			from: new Date( 'Wed Sep 19 2018 12:00:00' ),
			to: new Date( 'Fri Sep 21 2018 12:00:00' ),
		};

		store.dispatch( thunks.setDates( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	describe( 'setDateTime', () => {
		test( 'Set only the start date', () => {
			const attributes = {
				start: '2018-06-10 17:00:00',
			};
			store.dispatch( thunks.setDateTime( attributes ) );
			expect( store.getActions() ).toMatchSnapshot();
		} );

		test( 'Set the start and end date on the same day', () => {
			const attributes = {
				start: '2018-06-10 12:00:00',
				end: '2018-06-10 12:00:00',
			};
			store.dispatch( thunks.setDateTime( attributes ) );
			expect( store.getActions() ).toMatchSnapshot();
		} );

		test( 'Set the start and end date on different days', () => {
			const attributes = {
				start: '2018-05-01 12:00:00',
				end: '2018-05-04 20:00:00',
			};
			store.dispatch( thunks.setDateTime( attributes ) );
			expect( store.getActions() ).toMatchSnapshot();
		} );
	} );

	test( 'Set multi day to true', () => {
		const attributes = {
			start: '2018-06-05 17:00:00',
			end: '2018-06-05 17:30:00',
			isMultiDay: true,
		};

		store.dispatch( thunks.setMultiDay( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set multi day to false when end time is later than start time', () => {
		const attributes = {
			start: '2018-06-05 17:00:00',
			end: '2018-06-08 17:30:00',
			isMultiDay: false,
		};

		store.dispatch( thunks.setMultiDay( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set multi day to false when end time is earlier than start time', () => {
		const attributes = {
			start: '2018-06-05 17:00:00',
			end: '2018-06-08 15:30:00',
			isMultiDay: false,
		};

		store.dispatch( thunks.setMultiDay( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Set multi day to false when end time is earlier than start time', () => {
		const attributes = {
			start: '2018-06-05 23:30:00',
			end: '2018-06-08 15:30:00',
			isMultiDay: false,
		};

		store.dispatch( thunks.setMultiDay( attributes ) );

		expect( store.getActions() ).toMatchSnapshot();
	} );
} );
