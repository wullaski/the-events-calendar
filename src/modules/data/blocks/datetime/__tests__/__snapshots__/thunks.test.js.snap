// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Datetime thunks Initial set of state 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "June 5, 2018 5:00 pm",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "June 5, 2018 5:30 pm",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "allDay": false,
    },
    "type": "@@MT/EVENTS/SET_ALL_DAY",
  },
  Object {
    "payload": Object {
      "timeZone": "UTC",
    },
    "type": "@@MT/EVENTS/SET_TIME_ZONE",
  },
  Object {
    "payload": Object {
      "start": "2018-06-05 17:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-06-05 17:30:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "label": "June 5 2018 at 5:00 pm - 5:30 pm",
    },
    "type": "@@MT/EVENTS/SET_NATURAL_LANGUAGE_LABEL",
  },
  Object {
    "payload": Object {
      "multiDay": false,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks Set all day off 1`] = `
Array [
  Object {
    "payload": Object {
      "allDay": false,
    },
    "type": "@@MT/EVENTS/SET_ALL_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks Set all day on 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-06-05 00:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-06-05 23:59:59",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "allDay": true,
    },
    "type": "@@MT/EVENTS/SET_ALL_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks Set dates from date picker with multi day off 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-09-19 17:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-09-19 17:30:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
]
`;

exports[`[STORE] - Datetime thunks Set dates from date picker with multi day on 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-09-19 17:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-09-21 17:30:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
]
`;

exports[`[STORE] - Datetime thunks Set end time 1`] = `
Array [
  Object {
    "payload": Object {
      "end": "2018-06-05 18:00:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
]
`;

exports[`[STORE] - Datetime thunks Set multi day to false when end time is earlier than start time 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-06-05 17:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-06-05 18:00:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "multiDay": false,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks Set multi day to false when end time is earlier than start time 2`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-06-05 22:30:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-06-05 23:30:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "multiDay": false,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks Set multi day to false when end time is later than start time 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-06-05 17:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-06-05 17:30:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "multiDay": false,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks Set multi day to true 1`] = `
Array [
  Object {
    "payload": Object {
      "end": "2018-06-08 17:30:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "multiDay": true,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks Set start time 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-06-05 02:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
]
`;

exports[`[STORE] - Datetime thunks setDateTime Set only the start date 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-06-10 17:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-06-10 18:00:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "multiDay": false,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks setDateTime Set the start and end date on different days 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-05-01 12:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-05-04 20:00:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "multiDay": true,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;

exports[`[STORE] - Datetime thunks setDateTime Set the start and end date on the same day 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "2018-06-10 12:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "2018-06-10 13:00:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "multiDay": false,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;
